spec/src/main/asciidoc/repository.asciidoc

```diff
@@ -1,2 +1,217 @@
+
+[source,java]
+----
+@Repository
+public interface ProductRepository extends CrudRepository<Product, Long> {
+ @Query("THE QUERY EXPRESSION")
+ Optional<Product> findByName(String name);
+}
+----
+
+Jakarta Data also brings ```Param``` annotation to define a binder annotation, where such as the query expression, each vendor will express the syntax freely such as ```?```, ```@```, etc..
+
+[source,java]
+----
+@Repository
+public interface ProductRepository extends CrudRepository<Product, Long> {
+@Query("THE QUERY EXPRESSION")
+Optional<Product> findByName(@Param("name") String name);
+}
+----
+
+
+==== Query by Method
+
+The Query by method mechanism allows for creating query commands by conventions.
+
+E.g.:
+
+[source,java]
+----
+@Repository
+public interface ProductRepository extends CrudRepository<Product, Long> {
+
+  List<Product> findByName(String name);
+
+  List<Person> findByNameOrderByNameAsc(String name);
+
+  List<Person> findByNameOrderByNameDesc(String name);
+
+}
+----
+
+The parsing query method name has two parts: the subject and the property.
+
+The first part defines the query's subject or condition, and the second the condition value; both forms the predicate.
+
+A predicate can refer only to a direct property of the managed entity. We also have the option to handle entities with another class on them.
+
+Assume an Order entity has an Address with a Zipcode. In that case, the access is ```order.address.zipCode```.
+
+The resolution algorithm starts by interpreting the whole part (AddressZipCode) as the property and checks the domain class for a property with that name (uncapitalized). If the algorithm succeeds, it uses that property.
+
+[source,java]
+----
+@Repository
+public interface OrderRepository extends CrudRepository<Order, Long> {
+
+  Stream<Order> findByAddressZipCode(ZipCode zipCode);
+
+}
+----
+
+
+Although this should work for most cases, to resolve this ambiguity, you can use ```_``` inside your method name to manually define traversal points.
+
+[source,java]
+----
+@Repository
+public interface OrderRepository extends CrudRepository<Order, Long> {
+
+  Stream<Order> findByAddress_ZipCode(ZipCode zipCode);
+
+}
+----
+
+WARNING: Define as a priority following standard Java naming conventions, camel case,  using underscore as the last instance resource.
+
+===== Query methods Keywords
+
+The following table lists the subject keywords generally supported by Jakarta Data.
+
+|===
+|Keyword |Description
+
+|findBy
+|General query method returning the repository type.
+
+|deleteBy
+|Delete query method returning either no result (void) or the delete count.
+
+|countBy
+|Count projection returning a numeric result.
+
+|existsBy
+|Exists projection, returning typically a ```boolean``` result.
+|===
+
+Jakarta Data often support the following list of predicate keywords. It might not work because some keywords listed here might not be supported in a particular store.
```

Comment:
"Jakarta Data often support"
should be
"Jakarta Data implementations often support"